Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs

```diff
@@ -27,8 +28,16 @@ public class CodeGenerationDisplayDriver : ContentTypeDefinitionDisplayDriver
 
                     // Building the code for the type.
                     var name = model.Name;
-                    codeBuilder.AppendLine($"_contentDefinitionManager.AlterTypeDefinition(\"{name}\", type => type");
-                    codeBuilder.AppendLine($"    .DisplayedAs(\"{model.DisplayName}\")");
+
+                    // This can't be added to a Helpful Libraries extension method because one with the
+                    // public static StringBuilder AppendLineInvariant(
+                    //     this StringBuilder stringBuilder,
+                    //     [InterpolatedStringHandlerArgument("", "provider")] ref StringBuilder.AppendInterpolatedStringHandler handler)
+                    // signature causes the following error:
+                    // "'StringBuilderExtensions.AppendLineInvariant(StringBuilder, ref StringBuilder.AppendInterpolatedStringHandler)'
+                    // is not an instance method, the receiver cannot be an interpolated string handler argument
+                    codeBuilder.AppendLine(CultureInfo.InvariantCulture, $"_contentDefinitionManager.AlterTypeDefinition(\"{name}\", type => type");
+                    codeBuilder.AppendLine(CultureInfo.InvariantCulture, $"    .DisplayedAs(\"{model.DisplayName}\")");

```

Comment:
What about `public static StringBuilder AppendLineInvariant(this StringBuilder stringBuilder, FormattableString text)`?

Reply:
Unless we construct a `StringBuilder.AppendInterpolatedStringHandler` this won't work, since `FormattableString` needs to be converted to `StringBuilder.AppendInterpolatedStringHandler`

Reply:
Okay, I've looked into it and get it now. So StringBuilder does some spooky syntax sugar black magic for interpolated strings now. I don't think your original comment is clear about why you want to do that snippet, just that it doesn't work.
Based on [this article](https://blog.ladeak.net/posts/string-interpolation-stringbuilder-2#:~:text=currently%20it%20just%20returns%20the%20string%20builder%20without%20a%20side%20effect.) I don't think making that extension would do anything becuase it looks like that's just a signal for the compiler to do its weird special optimization.
Also the `"provider"` in `[InterpolatedStringHandlerArgument("", "provider")]` refers to the argument name that doesn't exist so that shouldn't work.

Originally I thought you'd just need to stringify the interpolation before passing it to AppendLine.
```csharp
public static StringBuilder AppendLineInvariant(this StringBuilder stringBuilder, FormattableString text) =>
    stringBuilder.AppendLine(text..ToString(CultureInfo.InvariantCulture));
...
codeBuilder.AppendLineInvariant($"_contentDefinitionManager.AlterTypeDefinition(\"{name}\", type => type");
```

This all seems to be way more work than what is worth. Leave the code, but please clarify in the comment what you were going for.

Reply:
Yep, it's crazy.

Done now.